scripts/core/api/api-service.js

```diff
@@ -246,6 +246,31 @@ function APIProvider() {
             });
         };
 
+        Resource.prototype.getAll = function(params) {
+            var p = $q.when();
+            var deferred = $q.defer();
```

Comment:
it's better to work with the promise you get from query, it would also handle errors in fetching and would reject the promise